src/fsharp/vs/service.fs

```diff
@@ -2642,38 +2673,43 @@ type BackgroundCompiler(referenceResolver, projectCacheSize, keepAssemblyContent
         | None -> parseCacheLock.AcquireLock (fun ctok -> parseAndCheckFileInProjectCachePossiblyStale.TryGet(ctok,(filename,options)))
 
     /// Parse and typecheck the whole project (the implementation, called recursively as project graph is evaluated)
-    member private bc.ParseAndCheckProjectImpl(options, ctok, ct) : FSharpCheckProjectResults =
-        let builderOpt,creationErrors,_ = getOrCreateBuilder (ctok, options, ct)
+    member private bc.ParseAndCheckProjectImpl(options, ctok) : Async<FSharpCheckProjectResults> =
+      async {
+        let builderOpt,creationErrors,_ = getOrCreateBuilder (ctok, options)
```

Comment:
I'm pretty sure that ``getOrCreateBuilder`` should end up being ``Async`` (and thus cancellable) in this new world?